File: java/src/main/java/com/ensta/gestion/Cours.java
package com.ensta.gestion;

/**
 * Cours académique — SRP : représente uniquement un cours
 * (nom + professeur responsable). Aucune logique métier autre.
 */
public class Cours {

    private final String nomCours;
    private final String professeurResponsable;

    public Cours(String nomCours, String professeurResponsable) {
        if (nomCours == null || nomCours.trim().isEmpty()) {
            throw new IllegalArgumentException("Le nom du cours ne peut pas être vide.");
        }
        if (professeurResponsable == null || professeurResponsable.trim().isEmpty()) {
            throw new IllegalArgumentException("Le professeur responsable ne peut pas être vide.");
        }
        this.nomCours = nomCours.trim();
        this.professeurResponsable = professeurResponsable.trim();
    }

    public String getNomCours() {
        return nomCours;
    }

    public String getProfesseurResponsable() {
        return professeurResponsable;
    }

    @Override
    public String toString() {
        return nomCours + " (Prof : " + professeurResponsable + ")";
    }
}

File: python/cours.py
"""Module Cours — Représente un cours académique.

SRP : cette classe a une seule responsabilité — représenter un cours
(nom + professeur responsable). Aucune logique d'affichage métier,
aucune persistance, aucune notification.
"""


class Cours:
    """Un cours académique avec un nom et un professeur responsable."""

    def __init__(self, nom_cours: str, professeur_responsable: str):
        if not nom_cours or not nom_cours.strip():
            raise ValueError("Le nom du cours ne peut pas être vide.")
        if not professeur_responsable or not professeur_responsable.strip():
            raise ValueError("Le professeur responsable ne peut pas être vide.")
        self.__nom_cours = nom_cours.strip()
        self.__professeur_responsable = professeur_responsable.strip()

    @property
    def nom_cours(self) -> str:
        return self.__nom_cours

    @property
    def professeur_responsable(self) -> str:
        return self.__professeur_responsable

    def __str__(self) -> str:
        return f"{self.__nom_cours} (Prof : {self.__professeur_responsable})"

    def __repr__(self) -> str:
        return f"Cours(nom_cours={self.__nom_cours!r}, professeur={self.__professeur_responsable!r})"
